andebox/actions/changelog_fragment.py

```diff
@@ -0,0 +1,335 @@
+
+    @staticmethod
+    def _get_base_branch(repo) -> str:
+        # Try common default branch names
+        for branch in ["main", "master"]:
+            try:
+                repo.commit(branch)
+                return branch
+            except Exception:
+                continue
+        return "HEAD~1"
+
+    @staticmethod
+    def _get_staged_unstaged_files(repo) -> Set[str]:
+        changed_files = set()
+
+        # Get unstaged changes
+        for item in repo.index.diff(None):
+            if item.a_path:
+                changed_files.add(item.a_path)
+            if item.b_path:
+                changed_files.add(item.b_path)
+
+        # Get staged changes
+        for item in repo.index.diff("HEAD"):
+            if item.a_path:
+                changed_files.add(item.a_path)
+            if item.b_path:
+                changed_files.add(item.b_path)
+
+        return changed_files
+
+    @staticmethod
+    def get_plugin_paths(context: ConcreteContext) -> List[str]:
+        if context.type == ContextType.COLLECTION:
+            return [
+                "plugins/modules/",
+                "plugins/module_utils/",
+                "plugins/lookup/",
+                "plugins/filter/",
+                "plugins/test/",
+                "plugins/callback/",
+                "plugins/connection/",
+                "plugins/inventory/",
+                "plugins/action/",
+                "plugins/cache/",
+                "plugins/become/",
+                "plugins/cliconf/",
+                "plugins/doc_fragments/",
+                "plugins/httpapi/",
+                "plugins/netconf/",
+                "plugins/shell/",
+                "plugins/strategy/",
+                "plugins/terminal/",
+                "plugins/vars/",
+            ]
+        elif context.type == ContextType.ANSIBLE_CORE:
+            return [
+                "lib/ansible/modules/",
+                "lib/ansible/module_utils/",
+                "lib/ansible/plugins/lookup/",
+                "lib/ansible/plugins/filter/",
+                "lib/ansible/plugins/test/",
+                "lib/ansible/plugins/callback/",
+                "lib/ansible/plugins/connection/",
+                "lib/ansible/plugins/inventory/",
+                "lib/ansible/plugins/action/",
+                "lib/ansible/plugins/cache/",
+                "lib/ansible/plugins/become/",
+                "lib/ansible/plugins/cliconf/",
+                "lib/ansible/plugins/doc_fragments/",
+                "lib/ansible/plugins/httpapi/",
+                "lib/ansible/plugins/netconf/",
+                "lib/ansible/plugins/shell/",
+                "lib/ansible/plugins/strategy/",
+                "lib/ansible/plugins/terminal/",
+                "lib/ansible/plugins/vars/",
+            ]
+        else:
+            return []
+
+    def is_plugin_file(self, file_path: str, context: ConcreteContext) -> bool:
+        plugin_paths = self.get_plugin_paths(context)
```

Comment:
That is not what I meant by "use the context class". The context is the object that is aware of the differences between a collection and ansible core. This kind of logic should be _in_ the context. But now I am scared shitless of what you will do there.